### ISSUE TITLE
A form resulting from a call to Form.ShowDialog() should be exposed programmatically as a dialog.

### ISSUE THREAD

**Issue by guybark** [dotnet/winforms#2907]
**Is your feature request related to a problem? Please describe.**
According to https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-automation-element-propids, UIA providers can set UIA_IsDialogPropertyId true on UI that's semantically a dialog. Based on that property, an assistive technology such as a screen reader can decide how best to convey the dialog's information to the customer. In the case of Narrator, today Narrator can decide to convey some contents of a dialog window when the window appears, beyond announcing only the focused control and the name of the dialog. When Form.ShowDialog() is called, the resulting window is not exposed as having a UIA IsDialog property of true, and so it is not being semantically exposed through UIA as being a dialog.

**Describe the solution you'd like**
The UIA Window element associated with the form that appears in response to the Form.ShowDialog() call has a UIA IsDialog property of true.

**Describe alternatives you've considered**
As far as I know, there's no straightforward and robust way for an app to get the IsDialog property set true on the form.

**Will this feature affect UI controls?**
Yes.
- Will VS Designer need to support the feature? If yes, describe how you expect it to funсtion.
No.
- What impact will it have on accessibility?
The form will programmatically be exposed as a dialog, which is the correct match for its semantics.
- Will this feature need to be localized or be localizable?
No.

**Comment by merriemcgaw** [comment]
You're more than welcome to submit a PR if you fancy 😉

**Comment by NikitaSemenovAkvelon** [comment]
In progress
